Neos.Flow.Log/Classes/PsrLogger.php

```diff
@@ -0,0 +1,114 @@
+<?php
+namespace Neos\Flow\Log;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Log\Backend\BackendInterface;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LoggerTrait;
+use Psr\Log\LogLevel;
+
+/**
+ * PSR-3 supporting logger.
+ *
+ * @api
+ */
+class PsrLogger implements LoggerInterface
+{
+    use LoggerTrait;
+
+    const LOGLEVEL_MAPPING = [
+        LogLevel::EMERGENCY => LOG_EMERG,
+        LogLevel::DEBUG => LOG_DEBUG,
+        LogLevel::INFO => LOG_INFO,
+        LogLevel::NOTICE => LOG_NOTICE,
+        LogLevel::WARNING => LOG_WARNING,
+        LogLevel::ERROR => LOG_ERR,
+        LogLevel::CRITICAL => LOG_CRIT,
+        LogLevel::ALERT => LOG_ALERT
+    ];
+
+    /**
+     * @var BackendInterface[]
+     */
+    protected $backends = [];
+
+    /**
+     * PsrLogger constructor.
+     *
+     * @param iterable $backends
+     */
+    public function __construct(iterable $backends)
+    {
+        foreach ($backends as $backend) {
+            if (!$backend instanceof BackendInterface) {
+                throw new \InvalidArgumentException(sprintf('The provided backend was not an instance of the "%s" interface', BackendInterface::class), 1515338089289);
+            }
+            $backend->open();
+            $this->backends[] = $backend;
+        }
+    }
+
+    /**
+     * @param mixed $level
```

Comment:
Is it really mixed? Weird…

Reply:
I think the interface declares it as mixed, that's where this comes from...